src/commands/executeFunction/eventGrid/EventGridFileOpenStep.ts

```diff
@@ -0,0 +1,95 @@
+import * as path from "path";
+import * as vscode from 'vscode';
+import { type Progress } from "vscode";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { feedUtils } from "../../../utils/feedUtils";
+import { type EventGridExecuteFunctionContext } from "./EventGridExecuteFunctionContext";
+
+export class EventGridFileOpenStep extends AzureWizardExecuteStep<EventGridExecuteFunctionContext> {
+    public priority: number;
+
+    public async execute(context: EventGridExecuteFunctionContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined; }>): Promise<void> {
+        const eventSource = nonNullProp(context, 'eventSource');
+        const selectedFileName = nonNullProp(context, 'selectedFileName');
+        const selectedFileUrl = nonNullProp(context, 'selectedFileUrl');
+
+        // Get selected contents of sample request
+        const downloadingMsg: string = localize('downloadingSample', 'Downloading sample request...');
+        progress.report({ message: downloadingMsg });
+        const selectedFileContent = await feedUtils.getJsonFeed(context, selectedFileUrl);
+
+        // Create a temp file with the sample request & open in new window
+        const openingFileMsg: string = localize('openingFile', 'Opening file...');
+        progress.report({ message: openingFileMsg });
+        const tempFilePath: string = await createTempSampleFile(eventSource, selectedFileName, selectedFileContent);
+        const document: vscode.TextDocument = await vscode.workspace.openTextDocument(tempFilePath);
+        await vscode.window.showTextDocument(document, {
+            preview: false,
+        });
+        ext.fileToFunctionNodeMap.set(document.fileName, nonNullProp(ext, 'currentExecutingFunctionNode'));
+        context.fileOpened = true;
+
+        // Request will be sent when the user clicks on the button or on the codelens link
+        // Show the message only once per workspace
+        if (!ext.context.workspaceState.get('didShowEventGridFileOpenMsg')) {
+            const doneMsg = localize('modifyFile', "You can modify the file and then click the 'Save and execute' button to send the request.");
+            void vscode.window.showInformationMessage(doneMsg);
+            await ext.context.workspaceState.update('didShowEventGridFileOpenMsg', true);
+        }
+
+
+        // Set a listener to delete the temp file after it's closed
+        void new Promise<void>((resolve, reject) => {
+            const disposable = vscode.workspace.onDidCloseTextDocument(async (closedDocument) => {
+                if (closedDocument.fileName === document.fileName) {
+                    try {
+                        ext.fileToFunctionNodeMap.delete(document.fileName);
+                        await AzExtFsExtra.deleteResource(tempFilePath);
+                        resolve();
+                    } catch (error) {
+                        reject(error);
```

Comment:
We probably just want to swallow the error here.

You can add this line to do so:
`context.errorHandling.suppressDisplay = true;`